src/engine/engine.ts

```diff
@@ -38,15 +49,17 @@ export class Engine {
   }
 
   async init() {
-    Array.from(this.backfillers.entries())
-      .map(([selector]) => selector)
-      .forEach(async (selector) => {
-        // @todo(notVitaliy): Fix this shit... eventually
-        const days = 5
-        await this.backfill(selector, days)
-        // this is fucked for now. Will only tick for the first strategy for a currency pair
-        this.tick(selector)
+    this.backfillers.forEach(async (backfillState, selector) => {
+      backfillState.status = BACKFILL_STATUS.RUNNING
+      await this.backfill(selector, backfillState.days)
+      backfillState.status = BACKFILL_STATUS.DONE
```

Comment:
Not sure if there's an explicit need for this since `this.backfill` will only return once it's done. It'd be safe to invoke `forEach (strat) -> this.tick` right after.

Reply:
you mean to drop the state? i am okay with that, i just added it because i found your enum.

But with the backfiller state, you could trace it p.e. for ui  if the backfiller is still in running state.

Reply:
It was a half-finished thought. Just don't really see a use case for it right now.

The plan for exposing data to the UI would be a singleton class. The service that can be imported where needed and then pipe data into it. The service would have some helper methods and hooks.

It'd mostly be a wrapper around a WS server.